neuralpp/symbolic/sympy_expression.py

```diff
@@ -264,7 +268,38 @@ def atom(self) -> Any:
         return SymPyExpression.pythonize_value(self._sympy_object)
 
 
-class SymPyFunctionApplication(SymPyExpression, FunctionApplication):
+class SymPyFunctionApplicationInterface(SymPyExpression, FunctionApplication, ABC):
```

Comment:
I created an interface because `SymPyCondFunctionApplication` is created for piecewise/cond operators specifically, which overloads `arguments`. The idea is we want a unified "if-then-else" conditional expression interface for all expressions (`expr.arguments[0]` would be the if-clause, `expr.argument[1]` the then-clause, `expr.argument[2]` the else-clause), but internal sympy uses `piecewise` (whose `args[0]` is `(then-clause, if-clause)`), so we need to do something under the hood to unify it to the interface (see `sympy_piecewise_to_if_then_else` and `sympy_piecewise_from_if_then_else` for detail).